spec/watirspec/browser_spec.rb

```diff
@@ -508,6 +508,34 @@
     end
   end
 
+  describe '#wait' do
+    not_compliant_on :chrome do
+      it 'waits for document ready state to be complete' do
+        @original = WatirSpec.implementation.clone
+        browser.close
+        @opts = WatirSpec.implementation.browser_args.last
+
+        @opts[:page_load_strategy] = 'none'
+        browser = WatirSpec.new_browser
+
+        # this is just an example of a site that takes a while to load
+        # TODO: Figure out a way to get a local site to take a set time to load; I couldn't figure it out with JS
+        browser.goto 'http://www.chicagotribune.com/'
```

Comment:
Given we are testing `#wait`, would it be bad to stub `#ready_state`? For example:

~~~~~~~~ruby
start_time = Time.now
allow(browser).to receive(:ready_state) { Time.now < start_time + 10 ? 'loading' : 'complete' }
expect(browser.ready_state).to eq 'loading'
browser.wait(20)
expect(browser.ready_state).to eq 'complete'
~~~~~~~~

Reply:
Alternatively, you can create a page with a lot of iframes (idea from https://javascript.info/onload-ondomcontentloaded). The following page has many iframes to forms_with_input_elements.html (our largest HTML file in the specs). For me, the page was in the "interactive" state for 4 seconds. You can increase the number of iframes if you want the wait to be longer.

~~~~~~~~html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
    </head>
    <body>
        <div id="events"></div>
        <script>
            // Source: https://javascript.info/onload-ondomcontentloaded
            function log(txt) {
                events.insertAdjacentHTML('beforeend', `<div>[${Math.floor(performance.now())}] ${txt}</div>`);
            }

            log('initial readyState:' + document.readyState);

            document.addEventListener('readystatechange', () => log('readyState:' + document.readyState));
            document.addEventListener('DOMContentLoaded', () => log('DOMContentLoaded'));

            window.onload = () => log('window onload');
        </script>

        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
        <iframe src="forms_with_input_elements.html" style="visibility:hidden" onload="log('iframe onload')"></iframe>
    </body>
</html>
~~~~~~~~

Reply:
I'm also using this method to test that `#ready_state` actually does something, so I didn't want to actually mock it out. Then again, now that I've proven it does something, maybe actually testing that part doesn't matter... I'll think on it before merging.